en/Search Algorithms/Floyd Cycle Detection Algorithm to Find Duplicate Number.md

```diff
@@ -5,34 +5,68 @@
 Given an array of integers containing `n + 1` integers, where each integer is in the range `[1, n]` inclusive. If there is only one duplicate number in the input array, this algorithm returns the duplicate number without modifying the original array, otherwise, it returns -1.
 
 ## Approach
+- We can imagine the array `arr` as a directed graph where each element is a node. Node with value `arr[i]` will point to node with value `arr[arr[i]]`.
```

Comment:
I find this confusing. I'd just say "`arr[i]` is the index of the node to which the i-th node points". Perhaps give a small example, like `[1, 0]`, and explain it.